### PR TITLE
Group name edition

### LINKED ISSUE #165: Group edition step 2: edit title
## Motivations
Be able to edit group

## Subtasks
* use this layout : https://france-ioi.github.io/algorea-designs/01.Groups_08.Edit_02_Edit-informations (but without everything but title ;-))
* let the user change the group name.
* on click on the save button in the header, display a success or error toast and redirect to the main page (same as in items)

### PR DESCRIPTION
**Pull request by GabrielVidal1**
Fixes #165 
I tried to make the layout responsive (so that the fields adapt to the width of the page) but I don't know if this is the right way to do it.

### CODE REVIEW

File: src/app/modules/group/http-services/group-update.service.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  updateGroup(
+    groupId: string,
+    changes: GroupChanges
+  ) : Observable<void> {
+    return this.http.put<SimpleActionResponse>(
+      `${appConfig().apiUrl}/groups/${groupId}`,
+      changes,
+    ).pipe(map(assertSuccess));
```
Comment (smadbe) [inline src/app/modules/group/http-services/group-update.service.ts:27]: Lots of line break, I don't think it makes things very readable (of course that's subjective):
```suggestion
  updateGroup(groupId: string, changes: GroupChanges) : Observable<void> {
    return this.http.put<SimpleActionResponse>(`${appConfig().apiUrl}/groups/${groupId}`, changes).pipe(
      map(assertSuccess),
    );
```

File: src/app/modules/group/pages/group-edit/group-edit.component.ts [hunk 2]
```diff
@@ -9,20 +17,88 @@ import { GroupDataSource } from '../../services/group-datasource.service';
...
     this.currentContent.editState.next('editing');
+
+    this.subscriptions.push(
+      this.state$.pipe(
+      filter<Ready<Group> | Fetching | FetchError, Ready<Group>>(isReady),
+    ).subscribe(state => {
+      this.groupId = state.data.id;
+      this.groupForm.patchValue({
+        name: state.data.name,
+      });
```
Comment (smadbe) [inline src/app/modules/group/pages/group-edit/group-edit.component.ts:46]: indent issue?

File: src/app/modules/group/pages/group-edit/group-edit.component.ts [hunk 3]
```diff
@@ -9,20 +17,88 @@ import { GroupDataSource } from '../../services/group-datasource.service';
...
+    if (this.groupForm.invalid) {
+      this.errorToast('You need to solve all the errors displayed in the form to save changes.');
+      return;
+    }
+
+    this.groupUpdateService.updateGroup(
+      this.groupId,
+      this.getGroupChanges(),
+    ).subscribe(
+      _status => {
```
Comment (smadbe) [inline src/app/modules/group/pages/group-edit/group-edit.component.ts:95]: what status? It is an `Observable<void>`

**Review by smadbe — CHANGES_REQUESTED** [review]
Good 👍  and seems to work :-) The responsiveness is also working (could be improved but that will be for later)

File: src/app/modules/group/pages/group-edit/group-edit.component.ts (hunk 3 above)
Comment (GabrielVidal1) [reply src/app/modules/group/pages/group-edit/group-edit.component.ts]: fixed

File: src/app/modules/group/pages/group-edit/group-edit.component.ts (hunk 2 above)
Comment (GabrielVidal1) [reply src/app/modules/group/pages/group-edit/group-edit.component.ts]: fixed

File: src/app/modules/group/http-services/group-update.service.ts (hunk 1 above)
Comment (GabrielVidal1) [reply src/app/modules/group/http-services/group-update.service.ts]: fixed

File: src/app/modules/group/pages/group-edit/group-edit.component.ts [hunk 4]
```diff
@@ -37,14 +37,12 @@ export class GroupEditComponent implements OnDestroy {
...
-      this.groupId = state.data.id;
-      this.groupForm.patchValue({
-        name: state.data.name,
-      });
-    }));
+      this.state$.pipe(filter<Ready<Group> | Fetching | FetchError, Ready<Group>>(isReady))
+      .subscribe(state => {
+        this.groupId = state.data.id;
+        this.groupForm.patchValue({ name: state.data.name });
+      })
```
Comment (smadbe) [inline src/app/modules/group/pages/group-edit/group-edit.component.ts:44]: no, that is still wrong 😕

File: src/app/modules/group/pages/group-edit/group-edit.component.ts [hunk 5]
```diff
@@ -90,13 +90,14 @@ export class GroupEditComponent implements OnDestroy {
       this.groupId,
       this.getGroupChanges(),
     ).subscribe(
-      _status => {
+      _result => {},
```
Comment (smadbe) [inline src/app/modules/group/pages/group-edit/group-edit.component.ts:93]: but what result? it is observable of void. isn't it?

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/group/pages/group-edit/group-edit.component.ts (hunk 4 above)
Comment (GabrielVidal1) [reply src/app/modules/group/pages/group-edit/group-edit.component.ts]: fixed

File: src/app/modules/group/pages/group-edit/group-edit.component.ts (hunk 5 above)
Comment (GabrielVidal1) [reply src/app/modules/group/pages/group-edit/group-edit.component.ts]: fixed

**Review by smadbe — APPROVED** [review]

**Pull request merged**